### PR TITLE
Use meta `NEW_COMPONENT` action to set component defaults

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR is a supporting change for [story #366938](https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/366938)

It enables the `Meta.NEW_COMPONENT` action to create components using [**component-types.ts** templates](https://github.com/DEFRA/forms-designer/blob/main/model/src/components/component-types.ts)

Also supports future work to:

* Organise the "Add component" page
* Pre-populate custom error messages where necessary
* Add default titles and hints to other components
* Add default options and schema configs (min, max etc)

## Email addresses
For example, default-but-editable email address hint text:

Email address default hint

### CODE REVIEW

File: designer/client/src/ComponentTypeEdit.tsx [hunk 1]
```diff
@@ -55,7 +52,8 @@ export const ComponentTypeEdit: FunctionComponent<Props> = (props) => {
     ? componentTypeEditors[selectedComponent.type]
     : undefined
 
-  const { type, subType } = component ?? {}
+  const defaults = getComponentDefaults(selectedComponent)
```
Comment (colinrotherham) [inline designer/client/src/ComponentTypeEdit.tsx:55]: We can't reliably pull `type` or `subType` from the form definition JSON 😣

Like lots of other places, we should refer to the defaults for these properties

File: designer/client/src/LinkCreate.test.tsx [hunk 2]
```diff
@@ -12,16 +13,17 @@ import React from 'react'
...
     {
       title: 'First page',
       path: '/first-page',
       components: [
         {
-          type: ComponentType.YesNoField,
           name: 'ukPassport',
           title: 'Do you have a UK passport?',
+          type: ComponentType.YesNoField,
+          subType: ComponentSubType.Field,
```
Comment (colinrotherham) [inline designer/client/src/LinkCreate.test.tsx:26]: I've updated all types and test fixtures to require `type` and `subType`

Previously saved JSON won't necessarily have these so `subType` is still optional via Joi

File: designer/client/src/components/ComponentCreate/ComponentCreate.test.tsx [hunk 3]
```diff
@@ -62,8 +66,15 @@ describe('ComponentCreate:', () => {
     const $textarea = await waitFor(() => screen.getByLabelText('Content'))
     const $button = await waitFor(() => screen.getByRole('button'))
 
+    // Ensure fields are empty
+    await act(() => userEvent.clear($input))
+    await act(() => userEvent.clear($textarea))
```
Comment (colinrotherham) [inline designer/client/src/components/ComponentCreate/ComponentCreate.test.tsx:71]: Defaults are now set so we need to clear these fields before testing validation

File: designer/client/src/components/ComponentCreate/ComponentCreate.tsx [hunk 4]
```diff
@@ -88,11 +88,16 @@ function useComponentCreate(props) {
...
+   * but replace default name with random ID
+   */
+  function handleCreate(component: ComponentDef) {
     dispatch({
-      type: Fields.EDIT_TYPE,
+      type: Meta.NEW_COMPONENT,
       payload: {
-        type: component.type
+        ...component,
+        name: randomId()
```
Comment (colinrotherham) [inline designer/client/src/components/ComponentCreate/ComponentCreate.tsx:100]: We now merge in defaults via `...component` when creating a new one

File: designer/client/src/reducers/component/componentReducer.meta.ts [hunk 5]
```diff
@@ -20,20 +21,16 @@ export function metaReducer(
...
+
     case Meta.NEW_COMPONENT:
-      return {
-        ...state,
-        selectedComponent: {
-          name: randomId(),
-          title: '',
-          schema: {},
-          options: { required: true }
-        }
```
Comment (colinrotherham) [inline designer/client/src/reducers/component/componentReducer.meta.ts:31]: These were the old defaults for every component

**Review by davidjamesstone — APPROVED** [review]

**Pull request merged**